test/helpers/utils.spec.ts

```diff
@@ -131,4 +142,121 @@ describe('redirectsForNext404Route', () => {
       { force: false, from: '/fr/test', status: 404, to: '/server/pages/fr/404.html' },
     ])
   })
+
+  it.only('returns static 404 redirects when LEGACY_FALLBACK_FALSE is not set', async () => {
+    const prerenderManifest: PrerenderManifest = {
+      ...basePrerenderManifest,
+      dynamicRoutes: {
+        "/getStaticProps/[id]": {
+          "routeRegex": "^/getStaticProps/([^/]+?)(?:/)?$",
+          "dataRoute": "/_next/data/build-id/getStaticProps/[id].json",
+          "fallback": false,
+          "dataRouteRegex": "^/_next/data/build\\-id/getStaticProps/([^/]+?)\\.json$"
+      },
+      },
+    }
+
+    const dynamicRoutes = [
+      {
+        "page": "/getStaticProps/[id]",
+        "regex": "^/getStaticProps/([^/]+?)(?:/)?$",
+        "routeKeys": {
+            "nextParamid": "nextParamid"
+        },
+        "namedRegex": "^/getStaticProps/(?<nextParamid>[^/]+?)(?:/)?$"
+    },
+    ]
+
+    const middleware = await getMiddleware(path.resolve('.next'))
```

Comment:
Does the middleware affect the tests in any way? If so consider tests with and without it.

Reply:
I think if we wanted to test the matchesMiddleware functionality then it would matter. For these tests, we can just set the middleware to default to an empty array so that we can reach the logic we need to test related to the env var.